README.md

```diff
@@ -52,38 +59,51 @@ async function updateDom() {
   element.appendChild(...);
   element.id = "...";
 
-  // Acquire a lock.
-  await element.displayLock.acquire();
+  // Acquire a lock, and allow it to be unlocked by the user agent when
+  // necessary.
+  await element.displayLock.acquire({ activatable: true });
 
   // Append the element to the DOM.
   document.body.appendChild(element);
 
-  // Now we can update the element, causing co-operative updates. After that
-  // resolves, we can commit the element making it visible to the user.
-  element.displayLock.update().then(() => {
+  // Now even if the element is appended to the tree we can do DOM updates,
+  // change style, etc. to it but it won't affect rendering.
+  element.style = "...";
+  element.appendChild(...);
+
+  // Calling update() will allow rendering to go update the style & layout
+  // values through co-operative updates that may last multiple frames.
+  // Note that this will not cause the element to be painted/rendered.
+  // After the calculations finish the promise it returns will resolve.
+  let updatePromise = element.displayLock.update();
+  updatePromise.then(() => {
+    // Calling commit() will cause the element to be rendered on the next frame.
```

Comment:
This isn't quite accurate. In a flex situation, there could be a lot of work to be done after commit that we could not do during update and it could take more than a frame to do it.

I think we should document clearly somewhere (maybe not here, maybe just reference it from here) that update does its best to calculate everything and make commit cheap but there are situations like flex where this is impossible.

Reply:
cc @vmpstr to make sure

Reply:
Fergal's comment is correct and should be noted.

Reply:
Added a note that it might still take a long time in cases like flex